dpath/__init__.py

```diff
@@ -45,7 +49,22 @@ def _split_path(path: Path, separator: Optional[str] = "/") -> Union[List[PathSe
     else:
         split_segments = path.lstrip(separator).split(separator)
 
-    return split_segments
+    final = []
+    for segment in split_segments:
+        if (options.DPATH_ACCEPT_RE_REGEXP and isinstance(segment, str)
+                and segment[0] == '{' and segment[-1] == '}'):
+            try:
+                rs = segment[1:-1]
+                rex = re.compile(rs)
+            except Exception as reErr:
+                print(f"Error in segment '{segment}' string '{rs}' not accepted"
```

Comment:
This library doesn't make use of standard streams to pass information to the user. Please use an exception object and raise it (you may declare a new one if needed in `exceptions.py`).

Reply:
Ok, also added test for readability of exception message. Typically:
~~~
In segment '{zz)bad}' string 'zz)bad' not accepted as re.regexp:
==>	unbalanced parenthesis at position 2
~~~